security/self-assessment.md

```diff
@@ -0,0 +1,228 @@
+
+- **Compliance with Kubernetes Security Standards:**  
+  Align with Kubernetes security best practices by integrating with Kubernetes’ RBAC and leveraging standard APIs for storage operations. This ensures that storage services provided by OpenEBS adhere to the security requirements of modern Kubernetes clusters.
+
+### Non-Goals
+
+- **Primary Database Functionality:**  
+  OpenEBS is not designed to function as a primary database or data processing system. Its focus is on providing persistent storage for Kubernetes workloads rather than managing or processing data directly.
+
+- **Comprehensive Data Security Solution:**  
+  While OpenEBS implements robust security measures for storage operations, it does not provide a complete data security solution. Additional security measures (e.g., network security, encryption at rest, end-to-end encryption) may be required based on user requirements.
+
+- **Broad Infrastructure Management:**  
+  OpenEBS does not manage the underlying network infrastructure, container runtime security, or host operating system hardening. Its scope is limited to storage provisioning and management within the Kubernetes environment.
+
+- **Supply Chain and Build Security:**  
+  This assessment does not cover aspects of the OpenEBS build process, dependency management, or container image supply chain security. It focuses solely on the runtime security posture of the storage operations.
+
+- **Advanced Data Processing or Analytics:**  
+  OpenEBS does not include integrated data processing, advanced query capabilities, or native analytics functions. Its role is to provide reliable and secure storage services, leaving data processing to external systems.
+
+## Self-assessment use
+
+This self-assessment is created by the OpenEBS team to perform an internal analysis of the project’s security. It is not intended to provide a security audit of OpenEBS, or function as an independent assessment or attestation of OpenEBS’s security health.
+
+This document serves to provide OpenEBS users with an initial understanding of OpenEBS’s security, where to find existing security documentation, OpenEBS plans for security, and general overview of OpenEBS security practices, both for development of OpenEBS as well as security of OpenEBS.
+
+This document provides the CNCF TAG-Security with an initial understanding of OpenEBS to assist in a joint-assessment, necessary for projects under incubation. Taken together, this document and the joint-assessment serve as a cornerstone for if and when OpenEBS seeks graduation and is preparing for a security audit.
+
+## Security Functions and Features
+
+- **RBAC & Access Control:**  
+  - Enforces Kubernetes RBAC for secure storage operations.
+  - Uses CSI volume modes and NVMe Reservations to guarantee exclusive volume access.
+
+- **Secure Communication:**  
+  - Utilizes secure HTTP/gRPC channels between control-plane and node components.
```

Comment:
hmm secure how?

Reply:
The REST doesn't have TLS. The gRPC layer doesn't have it also?

Reply:
Not the io-engine gRPC

Reply:
I've removed this bullet item for Secure Communication.